### DISCUSSION TITLE
Dry-run feature?

### DISCUSSION THREAD

**Discussion by sgtsquiggs** [golang-migrate/migrate#985] [Ideas]
I don't think every db could support this, but a lot could apply transformations and then rollback. Any way to achieve this currently? I'm using pgx/v5.

**Comment by oker1** [comment]
I'd prefer a different dry-run, outputting the sql statements that would be run so you can evaluate the risks (locking, replication lag, etc), and run manually or hand over to DBAs

https://documentation.red-gate.com/flyway/flyway-cli-and-api/concepts/dry-runs

**Reply by fizyk** [reply]
I'd love to have either that or a shortened form just listing pending migrations.
